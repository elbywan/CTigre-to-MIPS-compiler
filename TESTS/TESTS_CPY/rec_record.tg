type intlist = { hd : int, tl : intlist} in
type intarr = array of int in

/*var tab := intarr[5] of 0 in */

var l1 := intlist { hd = 1, tl = l2} 
and l2 := intlist { hd = 2, tl = l1} in

print("l1: ");
printint(l1.hd);print(" ");
printint(l1.tl.hd);print(" ");
printint(l1.tl.tl.hd);print(" ");
printint(l1.tl.tl.tl.hd);print(" ");



print("\n");
print("l2: ");
printint(l2.hd);print(" ");
printint(l2.tl.hd);print(" ");
printint(l2.tl.tl.hd);print(" ");
printint(l2.tl.tl.tl.hd);print(" ")


