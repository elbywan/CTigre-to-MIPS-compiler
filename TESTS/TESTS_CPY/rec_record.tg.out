l1: 1 2 1 2 
l2: 2 1 2 1 
